src/lib/FlightTasks/tasks/Auto/FlightTaskAuto.cpp

```diff
@@ -100,6 +100,44 @@ bool FlightTaskAuto::updateInitialize()
 	return ret;
 }
 
+bool FlightTaskAuto::updateFinalize()
+{
+	// All the auto FlightTasks have to comply with defined maximum yaw rate
+	// If the FlightTask generates a yaw or a yawrate setpoint that exceeds this value
+	// if will see its setpoint constrained here
+	_limitYawRate();
+	return true;
+}
+
+void FlightTaskAuto::_limitYawRate()
+{
+	if (PX4_ISFINITE(_yaw_setpoint) || PX4_ISFINITE(_yaw_sp_prev)) {
+		// Limit the rate of change of the yaw setpoint
+		float dy_max = math::radians(_param_mpc_yawrauto_max.get()) * _deltatime;
+
+		if (fabsf(_yaw_setpoint - _yaw_sp_prev) < M_PI_F) {
```

Comment:
Isn't it possible to directly wrap the inputs instead of generating different control paths here?

Reply:
Fixed in #11920